### PR TITLE
add selectAnnotation prop for mapview to programatically select pin

### PR DESCRIPTION
**Pull request by dchun**
Adding new prop type for mapview for auto select of pin which is available in ios native as selectAnnotation but not yet in react-native

### CODE REVIEW

**Comment by facebook-github-bot-6** [comment]
By analyzing the blame information on this pull request, we identified @nicklockwood and @jerolimov to be potential reviewers.

**Comment by javache** [comment]
Can this just be named `selected`? Since it's already part of the annotation type, you don't need to repeat that.

File: React/Views/RCTMapManager.m [hunk 1]
```diff
@@ -312,6 +312,19 @@ - (void)mapView:(RCTMap *)mapView annotationView:(MKAnnotationView *)view
   }
 }
 
+- (void)mapView:(RCTMap *)mapView didAddAnnotationViews:(NSArray *)views
+
```
Comment (javache) [inline React/Views/RCTMapManager.m]: nit: remove empty line

File: React/Views/RCTMapManager.m [hunk 2]
```diff
@@ -312,6 +312,19 @@ - (void)mapView:(RCTMap *)mapView annotationView:(MKAnnotationView *)view
...
+{   
+  for (MKAnnotationView* view in views) {
+    if ([view.annotation isKindOfClass:[RCTMapAnnotation class]]) {
+      RCTMapAnnotation *annotation = (RCTMapAnnotation *)view.annotation;
+      if (annotation.selectAnnotation) {
```
Comment (javache) [inline React/Views/RCTMapManager.m]: How will this update correctly when you unselect an annotation?

Comment (dchun) [inline React/Views/RCTMapManager.m]: Assuming that an unselect occurs by tapping outside the callout the [didDeselectAnnotationView](https://developer.apple.com/library/ios/documentation/MapKit/Reference/MKMapViewDelegate_Protocol/#//apple_ref/occ/intfm/MKMapViewDelegate/mapView:didDeselectAnnotationView:) is called which updates the annotation.

**Comment by dchun** [comment]
Good point. But just so there is no confusion and to keep in line with the way that the property is defined in mapkit I thought selectAnnotation was appropriate

**Comment by facebook-github-bot-6** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @caabernathy as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/add-selectannotation-prop-for-mapview-to-programatically-select-pin

Product Pains has been very useful in highlighting the top bugs and feature requests:
https://productpains.com/product/react-native?tab=top

Also, if this issue is a bug, **please consider sending a pull request with a fix**.

**Pull request closed without merging**